code/modules/reagents/chemistry/holder.dm

```diff
@@ -348,7 +349,8 @@ var/const/INGEST = 2
 	if(flags & REAGENT_NOREACT)
 		STOP_PROCESSING(SSobj, src)
 		return
-
+	for(var/thing in reagents_generated_per_cycle)
+		add_reagent(thing, reagents_generated_per_cycle[thing])
```

Comment:
looks alright!

Reply:
Do you think this should be done via signals? The vast majority of `reagents/process()` calls won't need this `for` loop